src/components/DataGridWithInfiniteScroll/DataGridWithInfiniteScroll.tsx

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) 2022 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import {useTable, Column} from "react-table";
+import {Spinner, SpinnerSize} from "@dellstorage/clarity-react/spinner/Spinner";
+import {Constants} from "components/DataGridWithInfiniteScroll/Constants";
+
+/**
+ * General component description :
+ * DataGridWithInfiniteScroll :
+ * Datagrids are for organizing large volumes of data that users can perform actions on.
+ */
+
+/**
+ * @param {rows} rows data
+ * @param {columns} column details
+ * @param {className} CSS class names
+ * @param {style} to add custom CSS styles
+ * @param {isLoading} if true then show loading icon before table is rendered
+ * @param {className} is to get a custom class to be applied or else default class is applied.
+ */
+type DataGridProps = {
+    rows: {[key: string]: any}[];
```

Comment:
Here used any for value not for key.

Reply:
Why?  we can always define a proper type and use it.  Any is dangerous as you don't get any help during debugging. 
In this case, the Rows would be returned to Users.  So we should be able to define a type and use it.  Same for Columns too.

Reply:
Removed any, and used react table type